src/Service/HelloRetailRecommendationService.php

```diff
@@ -0,0 +1,132 @@
+<?php declare(strict_types=1);
+
+namespace Helret\HelloRetail\Service;
+
+use Helret\HelloRetail\Service\Models\RecommendationContext;
+use Shopware\Core\Content\Category\CategoryEntity;
+use Shopware\Core\Content\Cms\DataResolver\CriteriaCollection;
+use Shopware\Core\Content\Product\ProductDefinition;
+use Shopware\Core\Content\Product\SalesChannel\SalesChannelProductEntity;
+use Shopware\Core\Framework\DataAbstractionLayer\Entity;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Struct\ArrayEntity;
+use Shopware\Core\System\SalesChannel\Aggregate\SalesChannelDomain\SalesChannelDomainEntity;
+use Shopware\Core\System\SalesChannel\Entity\SalesChannelRepository;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Core\Content\Product\ProductCollection;
+
+class HelloRetailRecommendationService
+{
+    private const STATIC_SEARCH_KEY = 'hello-retail-recommendations';
+
+    private const EXTRA_DATA = "extraData";
+    private const ENDPOINT = "recoms";
+
+    /**
+     * @param HelloRetailClientService $client
+     * @param EntityRepository $productRepository
```

Comment:
@wexoigi need to indicate the TEntityCollection, in this case <ProductCollection>